src/components/radio-button-group/radio-button-group.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+    Component,
+    Event,
+    EventEmitter,
+    h,
+    Prop,
+    State,
+    Watch,
+} from '@stencil/core';
+import { ListItem } from '../list/list-item.types';
+import { LimelListCustomEvent } from '@limetech/lime-elements';
+
+/**
+ * Radio Button Group
+ *
+ * The Radio Button component provides a convenient way to create a group of radio buttons
+ * from an array of options. Radio buttons allow users to select a single option from
+ * multiple choices, making them ideal for exclusive selections.
+ *
+ * :::note
+ * A single radio button is never useful in a UI. Radio buttons should always come in groups
+ * of at least 2 options where only one can be selected at a time.
+ * :::
+ *
+ * @exampleComponent limel-example-radio-button-group-basic
+ * @exampleComponent limel-example-radio-button-group-icons
+ * @exampleComponent limel-example-radio-button-group-multiple-lines
```

Comment:
I'd love to add some examples with striped rows (and perhaps those interactive styles) like what we have in `limel-list` 👇 
<img width="733" height="794" alt="image" src="https://github.com/user-attachments/assets/21077cf2-f448-4967-8bc5-d96015f4bd82" />
But since those are not applied using props, I can't. Or actually I can, but I'd rather not do it here, unless we add new props for these in `limel-list`. We can add these later in follow up PRs.